controllers/controllers_suite_test.go

```diff
@@ -371,3 +373,82 @@ func setupMachineCRDs() {
 		return ph.Patch(ctx, dummies.CSMachine1, patch.WithStatusObservedGeneration{})
 	}, timeout).Should(Succeed())
 }
+
+func setCSMachineOwnerCRD(owner *fakes.CloudStackMachineOwner, specReplicas, statusReplicas, statusReadyReplicas *int32, statusReady *bool) {
+	owner.Spec.Replicas = specReplicas
+	Ω(k8sClient.Create(ctx, owner)).Should(Succeed())
+	key := client.ObjectKey{Namespace: owner.Namespace, Name: owner.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, owner)
+	}, timeout).Should(BeNil())
+
+	Eventually(func() error {
+		owner.Status.Ready = statusReady
+		owner.Status.Replicas = statusReplicas
+		owner.Status.ReadyReplicas = statusReadyReplicas
+		return k8sClient.Status().Update(ctx, owner)
+	}, timeout).Should(BeNil())
+}
+
+// setCAPIMachineAndCSMachineCRDs creates a CAPI and CloudStack machine with an appropriate ownership ref between them.
+func setCAPIMachineAndCSMachineCRDs(CSMachine *infrav1.CloudStackMachine, CAPIMachine *clusterv1.Machine) {
+	//  Create them.
+	Ω(k8sClient.Create(ctx, CAPIMachine)).Should(Succeed())
+	Ω(k8sClient.Create(ctx, CSMachine)).Should(Succeed())
+
+	// Fetch the CS Machine that was created.
+	key := client.ObjectKey{Namespace: dummies.CSCluster.Namespace, Name: CSMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachine)
+	}, timeout).Should(BeNil())
+
+	// Fetch the CAPI Machine that was created.
+	key = client.ObjectKey{Namespace: dummies.ClusterNameSpace, Name: CAPIMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CAPIMachine)
+	}, timeout).Should(BeNil())
+
+	// Set ownerReference to CAPI machine in CS machine and patch back the CS machine.
+	Eventually(func() error {
+		ph, err := patch.NewHelper(dummies.CSMachine1, k8sClient)
+		Ω(err).ShouldNot(HaveOccurred())
+		dummies.CSMachine1.OwnerReferences = append(dummies.CSMachine1.OwnerReferences, metav1.OwnerReference{
+			Kind:       "Machine",
+			APIVersion: clusterv1.GroupVersion.String(),
+			Name:       CAPIMachine.Name,
+			UID:        "uniqueness",
+		})
+		return ph.Patch(ctx, CSMachine, patch.WithStatusObservedGeneration{})
+	}, timeout).Should(Succeed())
+}
+
+func setMachineOwnerReference(CSMachine *infrav1.CloudStackMachine, ownerRef metav1.OwnerReference) {
```

Comment:
Could these methods be abstracted to operate on `client.Objects` instead of the specific types? We kind of do that already in a few places in our reconciler base methods.